pkg/kubelet/pod_workers.go

```diff
@@ -901,16 +980,32 @@ func (p *podWorkers) completeTerminatingRuntimePod(pod *v1.Pod) {
 
 	klog.V(4).InfoS("Pod terminated all orphaned containers successfully and worker can now stop", "pod", klog.KObj(pod), "podUID", pod.UID)
 
-	// if a static pod is being tracked, forget it
-	delete(p.terminatingStaticPodFullnames, kubecontainer.GetPodFullName(pod))
-
 	if status, ok := p.podSyncStatuses[pod.UID]; ok {
 		if status.terminatingAt.IsZero() {
 			klog.V(4).InfoS("Pod worker was terminated but did not have terminatingAt set, likely programmer error", "pod", klog.KObj(pod), "podUID", pod.UID)
 		}
 		status.terminatedAt = time.Now()
 		status.finished = true
 		status.working = false
+
+		fullname := status.fullname
+		if p.startedStaticPodsByFullname[fullname] == pod.UID {
+			delete(p.startedStaticPodsByFullname, fullname)
+		}
+		l := 0
```

Comment:
This whole section can just be deleted if you perform this logic in https://github.com/kubernetes/kubernetes/pull/104743/commits/e77d81267795d914ef7ebab7e13c6a6baf986971#r720357165.  There should always be a pod worker for any "waiting fullname", and if there isn't that's a bug in pod worker lifecycle that has to get fixed.